### ISSUE TITLE
Experiment with edge detector

### ISSUE THREAD

**Issue by klokan** [moravianlibrary/imagesearch.mzk.cz#9]
For example Canny Edge Detector.

It is in OpenCV available for use in Java / C.

On command line experiments via ImageMagick?
convert -negate -canny 0x1+10%+30% big.jpg.3 big3-canny.jpg

Similar maps on MZK to test:
http://mapy.mzk.cz/mollova-sbirka/atlas-austriacus/XXII/

for example:
http://mapy.mzk.cz/mzk03/000/904/188/2619267549/
and 
http://mapy.mzk.cz/mzk03/000/904/256/2619267548/

or others like:
![big2](https://cloud.githubusercontent.com/assets/59284/6459618/8822fba8-c190-11e4-915b-5fdd32250f56.jpg)
![big3](https://cloud.githubusercontent.com/assets/59284/6459619/8aec26ac-c190-11e4-92ff-3939cc3a6509.jpg)
![big3-canny](https://cloud.githubusercontent.com/assets/59284/6459579/3d49d39a-c190-11e4-99b3-dd28db51a583.jpg)
![big2-canny](https://cloud.githubusercontent.com/assets/59284/6459578/3d4771f4-c190-11e4-9c6b-e507ad3743c3.jpg)

**Comment by klokan** [comment]
Comment on results please @dudaerich

**Comment by dudaerich** [comment]
I tried to use the Canny edge detector for finding similar historical maps from http://mapy.mzk.cz/mollova-sbirka/atlas-austriacus/XXII/. I applied the same procedure as for searching identical maps using by perceptual hashing (specifically DHash), but maps were transformed by Canny edge detector before they were indexing.

Querying of similar maps is done in such way that first 30 maps, which have the smallest hamming distance to the query map, are returned. Tests showed that this approach really works. Returned set contained all similar maps. Problem is that there were also other non-similar maps that had smaller hamming distance than similar maps and thus, by this way, we cannot say which maps are similar and which are not. However we can get set of candidates on similar maps.
